lib/fig/protocol/artifactory.rb

```diff
@@ -0,0 +1,391 @@
+# coding: utf-8
+
+require 'net/http'
+require 'uri'
+
+require 'fig/logging'
+require 'fig/network_error'
+require 'fig/package_descriptor'
+require 'fig/protocol'
+require 'fig/protocol/netrc_enabled'
+
+require 'artifactory'
+
+module Fig; end
+module Fig::Protocol; end
+
+class Fig::ArtifactoryClient
+  attr_reader :upload_url, :download_url
+
+  @upload_url = ''
+  @download_url = ''
+  @creds = ''
+
+  def initialize(options = {})
```

Comment:
This method and the two after it are both unimplemented. Is this still part of what's in progress?